src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String WRONG_INDEX = "The index is out off list.";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(WRONG_INDEX);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        Node<T> currentNode = findNodeByIndex(index);
+        if (index == 0) {
+            Node<T> insertedNode = new Node<>(null, value, currentNode);
+            currentNode.prev = insertedNode;
+            head = insertedNode;
+        } else {
+            Node<T> prevNode = currentNode.prev;
+            Node<T> insertedNode = new Node<>(prevNode, value, currentNode);
+            prevNode.next = insertedNode;
+            currentNode.prev = insertedNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> changedNode = findNodeByIndex(index);
+        T oldValue = changedNode.value;
+        changedNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> deletedNode = findNodeByIndex(index);
+        unlinkNode(index, deletedNode);
+        return deletedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        if (size == 0) {
+            return false;
+        }
```

Comment:
it is redundant because if size == 0 then for-loop will be skipped and you will return false